catalog/cataloger_diff.go

```diff
@@ -175,17 +319,152 @@ func getDiffDifferences(ctx context.Context, tx db.Tx, limit int, after string)
+	parentScanner := NewDBLineageScanner(tx, parentID, UncommittedID, &scannerOpts)
+	batch := newDiffResultsBatchWriter(tx, diffResultsTableName)
+	var parentEnt *DBScannerEntry
+	records := 0
+	for childScanner.Next() {
+		// stop on limit
+		if limit > -1 && records >= limit {
+			break
+		}
+
+		// is child element is relevant
+		childEnt := childScanner.Value()
+		if !childEnt.ChangedAfterCommit(effectiveCommits.ChildEffectiveCommit) {
+			continue
+		}
+
+		// get next parent - next parentEnt that path >= child
+		parentEnt, err = ScanDBEntryUntil(parentScanner, parentEnt, childEnt.Path)
+		if err != nil {
+			return err
+		}
+
+		diffType := evaluateFromChildElementDiffType(effectiveCommits, parentID, childEnt, parentEnt)
+		if diffType == DifferenceTypeNone {
+			continue
+		}
+
+		err = batch.Write(&diffResultRecord{
+			SourceBranch: childID,
+			DiffType:     diffType,
+			Entry:        childEnt.Entry,
+			EntryCtid:    &childEnt.RowCtid,
+		})
+		if err != nil {
+			return err
+		}
+		records++
+	}
+	if err := childScanner.Err(); err != nil {
+		return err
+	}
+	return batch.Flush()
+}
+
+func createDiffResultsTable(ctx context.Context, executor sq.Execer) (string, error) {
+	diffResultsTableName, err := diffResultsTableNameFromContext(ctx)
+	if err != nil {
+		return "", err
+	}
+	_, err = executor.Exec("CREATE UNLOGGED TABLE " + diffResultsTableName + ` (
+		source_branch bigint NOT NULL,
+		diff_type integer NOT NULL,
+		path character varying COLLATE "C" NOT NULL,
+		entry_ctid tid
+	)`)
+	if err != nil {
+		return "", fmt.Errorf("diff from child diff result table: %w", err)
+	}
+	return diffResultsTableName, nil
+}
+
+func evaluateFromChildElementDiffType(effectiveCommits *diffEffectiveCommits, parentBranchID int64, childEnt *DBScannerEntry, parentEnt *DBScannerEntry) DifferenceType {
+	var matchedParent *DBScannerEntry
+	if parentEnt != nil && parentEnt.Path == childEnt.Path {
+		matchedParent = parentEnt
+	}
+	// when the entry was deleted
+	if childEnt.IsDeleted() {
+		if matchedParent == nil || matchedParent.IsDeleted() {
+			return DifferenceTypeNone
+		}
+		if matchedParent.BranchID == parentBranchID {
+			// check if parent did any change to the entity
+			if matchedParent.MinCommit > effectiveCommits.ParentEffectiveCommit {
+				return DifferenceTypeConflict
+			}
+			return DifferenceTypeRemoved
+		}
+		// check if the parent saw this entry at the time
+		if matchedParent.MinCommit >= effectiveCommits.ParentEffectiveCommitByBranchID(matchedParent.BranchID) {
+			return DifferenceTypeRemoved
+		}
+		return DifferenceTypeNone
+	}
+	// when the entry was not deleted
+	if matchedParent != nil {
+		if matchedParent.IsDeleted() {
+			return DifferenceTypeAdded
+		}
+		if matchedParent.BranchID == parentBranchID {
+			// check if parent did any change to the entity
+			if matchedParent.MinCommit > effectiveCommits.ParentEffectiveCommit {
+				return DifferenceTypeConflict
+			}
+			return DifferenceTypeChanged
+		}
```

Comment:
I think else would make it more clear.

Reply:
what do you mean - anding 'else ' before each 'if'?